java/arcs/core/data/proto/ClaimProtoDecoder.kt

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.core.data.proto
+
+import arcs.core.data.Claim
+import arcs.core.data.HandleConnectionSpec
+
+/** Decodes an [ClaimProto] into [Claim]. */
+fun ClaimProto.decode(
+    particleSpecName: String,
+    connectionSpecs: Map<String, HandleConnectionSpec>
+) = when (claimCase) {
```

Comment:
Nit: don't use a function expression here, since this whole thing doesn't fit on one line

```
): Claim {
    return when (claimCase) {
        ...
    }
}
```

Reply:
Done.